spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2772,37 +2773,59 @@ private RuntimeException doInvokeRecordListener(final ConsumerRecord<K, V> cReco
 					DefaultKafkaListenerObservationConvention.INSTANCE,
 					() -> new KafkaRecordReceiverContext(cRecord, getListenerId(), this::clusterId),
 					this.observationRegistry);
-			return observation.observe(() -> {
+			RuntimeException runtimeException;
+			observation.start();
```

Comment:
I don't think this is what we need here.
There is just enough to pull `try..catch` from that supplier in the `observe()` around this `observe()`.

Reply:
Fix it, hope i get your point, and the code is much cleaner.